admin-dev/themes/new-theme/js/pages/product/components/categories/index.js

```diff
@@ -79,16 +83,8 @@ export default class CategoriesManager {
   }
 
   initTree() {
-    const initialElements = {};
-
-    this.categoryTree.querySelectorAll(ProductCategoryMap.treeElement).forEach((treeElement) => {
```

Comment:
That part you removed is what will give you the current product's categories, the idea is that on Add button click the manager is responsible for parsing the form data, so it loops through the tags and gets the list of categories IDs